### PR TITLE
Cubic Flat layer fixer

### PR DESCRIPTION
**Pull request by Foghrye4**
Screenshot from 2019-03-16 20-34-13
screen1
MalisisCore `UISelect` sux.

### CODE REVIEW

**Comment by Niko-sk2x** [comment]
How does that handle 2 layers with different biome in the same cube?

The multilayer superflat GUI is hard to use as it is. Manually typing height ranges into text fields where it doesn't fit. Lack of "-infinity" option. The blockstate selection GUI being nearly impossible to understand system, mixed together with layers (somehow?).

If you are going to improve it, re-do it in a better way. This time I'm not going to accept something like the current code that it is. I keep seeing a pattern where your code mostly works, but is nearly impossible to understand,  modify and fix by anyone else. Things that shouldn't be linked together, are linked in a way that is almost impossible to separate without rewriting it. Or maybe I just can't understand other people's code. I have no way to know, because there is noone else to review it.

File: src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/flat/FlatLayerTypeGsonAdapter.java [hunk 1]
```diff
@@ -45,59 +47,46 @@ public void write(JsonWriter out, Layer value) throws IOException {
...
+        int toY = Integer.MAX_VALUE;
+        IBlockState blockState = Blocks.STONE.getDefaultState();
+        Biome biome = null;
+        while (reader.hasNext()) {
+            String name = reader.nextName();
+            if (name.equals("fromY"))
+                fromY = reader.nextInt();
+            else if (name.equals("toY"))
+                toY = reader.nextInt();
+            else if (name.equals("blockstate")) {
```
Comment (Niko-sk2x) [inline src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/flat/FlatLayerTypeGsonAdapter.java]: How is this compatible with old saves? I don't see any relevant data fixer or similar thing.

Comment (Foghrye4) [reply src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/flat/FlatLayerTypeGsonAdapter.java]: Indeed. It is incompatible.

**Comment by Foghrye4** [comment]
Redone. Now contain only Json settings fixer. Other things - later.

**Comment by Niko-sk2x** [comment]
Later being "in a different PR" or "later in this PR"?

**Comment by Foghrye4** [comment]
Later in different PR.

**Pull request merged**